app/views/govuk_publishing_components/components/_radio.html.erb

```diff
@@ -1,26 +1,36 @@
 <%
   id_prefix ||= "radio-#{SecureRandom.hex(4)}"
   items ||= []
+  is_conditional = false
 %>
+<% if items.any? %>
+<% items.each do |item| %>
+  <% is_conditional = true if item["conditional"].present? %>
+<% end %>
+<div class="govuk-form-group">
+  <div class="govuk-radios" <%= 'data-module=radios' if is_conditional %>>
+<% end %>
 <% items.each_with_index do |item, index| %>
   <% if item === :or %>
-    <span class="gem-c-radio__block-text">
+    <div class="govuk-radios__divider">
       <%= t('components.radio.or') %>
-    </span>
+    </div>
   <% else %>
     <%
       item_next = items[index + 1] unless index === items.size - 1
       label_id = item[:id] ? item[:id] : "#{id_prefix}-#{index}"
       label_hint_id = "label-hint-#{SecureRandom.hex(4)}" if item[:hint_text].present?
+      conditional_id = "conditional-#{SecureRandom.hex(4)}" if item[:conditional].present?
     %>
     <div
       class="
         gem-c-radio
+        govuk-radios__item
```

Comment:
This stacking approach to individual items (in this case classes) leads to more readable code but puts extra whitespace in the outputted markup, which over time is going to (fractionally) increase our HTML file size. An alternative approach could be to assemble the classes needed outside of the markup and then insert them, e.g.

```
<%
  classes = "gem-c-radio govuk-radios__item"
  classes << " gem-c-radio--margin-bottom-0" if item_next === :or && item[:hint_text].present?
%>

<input class="<%= classes %>">
```

Reply:
Updated as suggested.